src/Orleans/Serialization/RemoteNonDeserializableException.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Runtime.Serialization;
+using Orleans.Runtime;
+
+namespace Orleans.Serialization
+{
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// Represents an exception which cannot be fully deserialized.
+    /// </summary>
+    [Serializable]
+    public class RemoteNonDeserializableException : OrleansException
```

Comment:
I don't think this should have anything internal in it as it prevents other external serializers from developing comparable feature sets.